tests/test_uninstall.py

```diff
@@ -0,0 +1,405 @@
+import os
+import sys
+from pathlib import Path
+from shutil import rmtree
+from subprocess import SubprocessError
+
+import pytest
+from conda.base.constants import COMPATIBLE_SHELLS
+from conda.common.path import win_path_to_unix
+from conda.core.initialize import (
+    _read_windows_registry,
+    make_initialize_plan,
+    run_plan,
+    run_plan_elevated,
+)
+from conftest import _get_shortcut_dirs, menuinst_pkg_specs, run_conda
+from ruamel.yaml import YAML
+
+ON_WIN = sys.platform == "win32"
+ON_MAC = sys.platform == "darwin"
+ON_LINUX = not (ON_WIN or ON_MAC)
+ON_CI = bool(os.environ.get("CI"))
+CONDA_CHANNEL = os.environ.get("CONDA_STANDALONE_TEST_CHANNEL", "conda-forge")
+
+
+@pytest.fixture(scope="function")
+def mock_system_paths(
+    monkeypatch,
+    tmp_path: Path,
+) -> dict[str, Path]:
+    paths = {}
+    if ON_WIN:
+        homedir = tmp_path / "Users" / "user"
+        monkeypatch.setenv("USERPROFILE", str(homedir))
+        # Monkeypatching LOCALAPPDATA will not help because user_cache_dir
+        # typically does not use environment variables
+        cachehome = homedir / "AppData" / "Local"
+    elif ON_MAC:
+        homedir = tmp_path / "Users" / "user"
+        cachehome = homedir / "Library" / "Caches"
+        monkeypatch.setenv("HOME", str(homedir))
+    else:
+        homedir = tmp_path / "home" / "users"
+        monkeypatch.setenv("HOME", str(homedir))
+        cachehome = homedir / "cache"
+
+    paths = {
+        "home": homedir,
+        "binstar": homedir / "ContinuumIO" / "binstar",
+        "cachehome": cachehome,
+        "confighome": homedir / "config",
+        "datahome": homedir / "data",
+    }
+    for mockdir in paths.values():
+        mockdir.mkdir(parents=True, exist_ok=True)
+    paths["baseenv"] = paths["home"] / "baseenv"
+
+    monkeypatch.setenv("XDG_CONFIG_HOME", str(paths["confighome"]))
+    monkeypatch.setenv("XDG_CACHE_HOME", str(paths["cachehome"]))
+    monkeypatch.setenv("XDG_DATA_HOME", str(paths["datahome"]))
+    monkeypatch.setenv("BINSTAR_CONFIG_DIR", str(paths["binstar"]))
+
+    return paths
+
+
+def create_env(
```

Comment:
I guess we are not using stuff from `conda.testing` like `tmp_env` because we want to handle the removal ourselves, right?

Reply:
There is a much simpler reason: I didn't have `conda` as a test dependency when I wrote that function and it was too small of a function for me to import `conda`.

However, since I need it for other things now, I replaced almost instances with the conda fixture. I had to keep `run_conda` for `menuinst` though because I cannot monkeypatch the paths otherwise.